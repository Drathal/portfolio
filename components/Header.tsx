import Link from 'next/link'
import { FC } from 'react'

import { makeStyles, Theme, createStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Fab from '@material-ui/core/Fab'
import Zoom from '@material-ui/core/Zoom'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp'
import useScrollTrigger from '@material-ui/core/useScrollTrigger'

interface Props {
  window?: () => Window
  children: React.ReactElement
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    title: {
      flexGrow: 1
    },
    appBar: {
      backgroundColor: 'rgba(0,0,0,0.8)'
    },
    rootZoom: {
      position: 'fixed',
      bottom: theme.spacing(2),
      right: theme.spacing(2)
    }
  })
)

function ScrollTop(props: Props) {
  const { children, window } = props
  const classes = useStyles()
  const trigger = useScrollTrigger({
    target: window ? window() : undefined,
    disableHysteresis: true,
    threshold: 100
  })

  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    const anchor = (
      (event.target as HTMLDivElement).ownerDocument || document
    ).querySelector('#back-to-top-anchor')

    if (anchor) {
      anchor.scrollIntoView({ behavior: 'smooth', block: 'center' })
    }
  }

  return (
    <Zoom in={trigger}>
      <div
        onClick={handleClick}
        role="presentation"
        className={classes.rootZoom}
      >
        {children}
      </div>
    </Zoom>
  )
}

const Header: FC = () => {
  const classes = useStyles()
  return (
    <>
      <nav className="nav">
        <AppBar
          classes={{
            root: classes.appBar
          }}
        >
          <Toolbar>
            <Typography variant="h6" className={classes.title}>
              <Link href="/">
                <a>Markus Dethlefsen</a>
              </Link>
            </Typography>

            <Button color="inherit">
              <Link href="/">
                <a>Timeline</a>
              </Link>
            </Button>

            <Button color="inherit">
              <Link href="/about">
                <a>About</a>
              </Link>
            </Button>
          </Toolbar>
        </AppBar>
        <Toolbar id="back-to-top-anchor" />
      </nav>

      <ScrollTop>
        <Fab color="secondary" size="small" aria-label="scroll back to top">
          <KeyboardArrowUpIcon />
        </Fab>
      </ScrollTop>
    </>
  )
}

export default Header
