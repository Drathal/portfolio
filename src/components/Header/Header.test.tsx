import { render } from '@testing-library/react'
import Header from './Header'

describe('<Header />', () => {
  test('it can render', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
