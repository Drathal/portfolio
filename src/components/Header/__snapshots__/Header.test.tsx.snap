// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it can render 1`] = `
<nav
  class="nav"
>
  <header
    class="MuiPaper-root MuiAppBar-root makeStyles-appBar-2 MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-4 MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        >
          <a
            href="/"
          >
            Markus Dethlefsen
          </a>
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              href="/"
            >
              Timeline
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              href="/about"
            >
              About
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    id="back-to-top-anchor"
  />
</nav>
`;
